### DISCUSSION TITLE
Work with NavigationRail

### DISCUSSION THREAD

**Discussion by PierreFaraut** [flet-dev/flet#282] [Q&A]
Hi there!
I'm trying to understand how to use NavigationRail class, and therefore navigate between pages. 
I'm trying to do a menu that will automatically display my pages, and then instantiate my page classes when needed. 
I come up with this code for the moment: 
```python
import flet
from flet import (
    Column,
    NavigationRail,
    Page,
    Row,
    VerticalDivider
)
from pages.home import Home
from pages.config import Config

def main(page: Page):

    def load_page(e):
        i = 0
        for a_page in list_pages:
            if i == e.control.selected_index:
                list_pages[i].visible = True
            else:
                list_pages[i].visible = False
            i += 1
        page.update()

    list_pages = [
        Home(),  # UserControl
        Config() # UserControl 
    ]

    rail = NavigationRail(
        selected_index=0,
        label_type="all",
        min_width=100,
        min_extended_width=400,
        group_alignment=-0.9,
        destinations=[a_page.nav_desti for a_page in list_pages], # Retrieve NavigationRailDestination from custom class
        on_change=load_page 
    )

    page.add(
        Row(
            [
                rail,
                VerticalDivider(width=1),
                Column(list_pages),
            ],
            expand=True,
        )
    )

flet.app(target=main)
```
The menu displays well the classes but:
- on initiation, nothing is displayed
- on initiation, all pages are rendered in background (which I don't necessarily want to). 
Any ideas on how to fix these issues? Do you have a better way to use the NavigationRail to display pages?
Thanks for your help! 
Pierre

**Comment by saurabhwadekar** [comment]
You can try using flet-route it will be easy for https://github.com/saurabhwadekar/flet_route

**Comment by carlos-alberto-code** [comment]
I solved it by adding the controls to the page, but I had to create structures to take care of the content change. You can see my project at https://github.com/carlos-alberto-code/punto-de-venta-uvm. In the views folder there is a module called AppView.py, there I declared in a view all the main content, but in a folder called naveasy I declared the structure that will help me to navigate and display the content.

**Comment by burhansvural** [comment]
@PierreFaraut 
Check out the codes in the file I added.
[ders_001_navigation_bar.tar.gz](https://github.com/user-attachments/files/16102034/ders_001_navigation_bar.tar.gz)
